### ISSUE TITLE
Windows build fails

### ISSUE THREAD

**Issue by iBotPeaches** [iBotPeaches/Apktool#1553]
The new License plugin has issues on Windows. My guess is newlines - https://github.com/hierynomus/license-gradle-plugin/issues/68

Lets submit an upstream PR to get this resolved. If we cannot, lets detect OS in build.gradle, if Windows, skip the license checks.

**Comment by iBotPeaches** [comment]
This is additionally blocked by

```

brut.androlib.VectorDrawableTest > checkIfDrawableFileDecodesProperly FAILED
    java.lang.IllegalArgumentException: Property 'http://javax.xml.XMLConstants/property/accessExternalDTD' is not recognized.
        at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl.setAttribute(DocumentBuilderFactoryImpl.java:116)
        at brut.androlib.res.xml.ResXmlPatcher.loadDocument(ResXmlPatcher.java:267)
        at brut.androlib.res.xml.ResXmlPatcher.removeManifestVersions(ResXmlPatcher.java:203)
        at brut.androlib.res.AndrolibResources.decodeManifestWithResources(AndrolibResources.java:220)
        at brut.androlib.Androlib.decodeManifestWithResources(Androlib.java:137)
        at brut.androlib.ApkDecoder.decode(ApkDecoder.java:113)
        at brut.androlib.VectorDrawableTest.checkIfDrawableFileDecodesProperly(VectorDrawableTest.java:54)

28 tests completed, 14 failed
```

So we have the issue with XMLConstants not available in certain environments, and builds are failing because license checks are running which for some reason aren't compatible in Windows. Staging that investigation for latter.

```
diff --git a/build.gradle b/build.gradle
index 472d4a6..cd89b01 100644
--- a/build.gradle
+++ b/build.gradle
@@ -53,8 +53,9 @@ allprojects {
         ext.ibot = 'Connor Tumbleson'
         ext.ibotEmail = 'connor.tumbleson@gmail.com'
     }
-    compileJava.finalizedBy licenseFormatMain
-    licenseFormatMain.finalizedBy licenseFormatTest
+	
+    gradle.startParameter.excludedTaskNames += "licenseMain"
+    gradle.startParameter.excludedTaskNames += "licenseTest"
 }
 
 if (!('release' in gradle.startParameter.taskNames)) {
@@ -94,6 +95,8 @@ build.doFirst {
 
 // used for official releases only. Please don't use
 task release {
+    compileJava.finalizedBy licenseFormatMain
+    licenseFormatMain.finalizedBy licenseFormatTest
 }
 ```

**Comment by bovirus** [comment]
@iBotPeaches 

Thanks for the patch.
Now pass the licence check biut I got aother error during the etsts

> Task :brut.apktool:apktool-cli:compileJava
Note: D:\GIT\Apktool\brut.apktool\apktool-cli\src\main\java\brut\apktool\Main.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :brut.apktool:apktool-lib:test

brut.androlib.UnknownDirectoryTraversalTest > invalidRootFileTest FAILED
    java.lang.AssertionError: Expected exception: brut.common.RootUnknownFileException

95 tests completed, 1 failed, 1 skipped

Thanks.

**Comment by iBotPeaches** [comment]
@bovirus The patch was not aimed to fix anything. It was a stash of my work so I could switch computers. Everything you posted above is warnings anyway. The error, I imagine, is the one I already posted. Please just wait until this bug is fixed. I'm well aware the build is failing.

**Comment by bovirus** [comment]
@iBotPeaches 

The liecense problems seems fixe.
After the update of 12.07.2015 compilation was ok but failed during tests.

> Task :brut.apktool:apktool-cli:compileJava
Note: D:\GIT\Apktool\brut.apktool\apktool-cli\src\main\java\brut\apktool\Main.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :brut.apktool:apktool-lib:test

brut.androlib.UnknownDirectoryTraversalTest > invalidRootFileTest FAILED
    java.lang.AssertionError: Expected exception: brut.common.RootUnknownFileException

95 tests completed, 1 failed, 1 skipped

**Comment by iBotPeaches** [comment]
It seems `accessExternalDTD` was a new property added in JAXP 1.5 - https://docs.oracle.com/javase/tutorial/jaxp/properties/

They advise checking if the property is supported - https://docs.oracle.com/javase/tutorial/jaxp/properties/

This is happening in other places
https://github.com/DaisyDiff/DaisyDiff/issues/7
https://github.com/gocd/gocd/issues/2040

Needs more research. I don't want to disable this property on Windows and asking the user base to use some specific java version and patch update is not right either. Need to patch in an updated xerces build which might resolve this.

**Comment by iBotPeaches** [comment]
Can't get any similar patches to work. I'm blaming Windows, so skipping any host system that doesn't have support for the new JAXP 1.5 features - https://github.com/iBotPeaches/Apktool/pull/1554

Not the fix I like, but it'll work.

**Comment by bovirus** [comment]
Also with latest patch (8 hours ago), I got the same error during final tests

brut.androlib.UnknownDirectoryTraversalTest > invalidRootFileTest FAILED
java.lang.AssertionError: Expected exception: brut.common.RootUnknownFileException

95 tests completed, 1 failed, 1 skipped

Could you pelase disable this test under window to make the compilation possible?

**Comment by iBotPeaches** [comment]
Building on Windows is on hold. I will wait till TravisCI supports Windows as I cannot continue to patching something I cannot regression test
